### DISCUSSION TITLE
Regarding the upcoming migration to native HA ble integration

### DISCUSSION THREAD

**Discussion by hugalafutro** [custom-components/ble_monitor#972] [General]
Hi,
I've read in the recent changelog that (long story short) HA will do what this integration does natively in upcoming release. On one hand I'm excited about new functionality in ha, on other I'm mortally terrified of even thinking about renaming some 40 entities provided (for now) by this integration, not to mention trying to remember which scripts/automation used which entity (since entity names straight up disappear from the automation/script ui once non-existent).

Will there be any way to migrate to native ha ble integration once the time comes in such a way that the entity names would not change?

**Comment by Ernst79** [comment]
Im afraid not, i think HA devs will use the new naming confension they came up with, which can be found in the development docs. But ble monitor will still work after the new Bluetooth implementation in ha. But im afraid, in the end there will be a time you will have to make the move. In the end, i think, you will be better of, as more maintainers will be available with way more experience in programming than me. There are already some improvements in the ble parsers made by the devs.

**Reply by hugalafutro** [reply]
Well if that's the price of progress I can adapt. Just to confirm I'm not missing some easier way to do this what I'll have to do is: 

- I'll install the new HA ble integration once it's live while passive ble still runs, add device to ha ble integration thus have 2 differently named entities for the same sensor.
- Remove the device from passive ble, shutdown ha.
- Then I go through all my automations & scripts .yamls with text editor and replace old sensor name with new name.
- Restart ha and hopefully everything works as if nothing changed.
- Then I just repeat the same for every device and regret the number of them I got 😄.

Last question I'd have is, will the native ha ble integration offer tracking of the devices? I have few keychain beacons I track via your integration and depend on them for home/away zone switching.

**Reply by Ernst79** [reply]
My first recommendation, first make sure every ble device you have is supported in the new integrations.  Only after you make sure they all are, make the move. They will support device tracking, but not sure if that will be in 2022.8 already. Till then, keep on using ble monitor.

**Reply by borpin** [reply]
> In the end, i think, you will be better of, as more maintainers will be available with way more experience in programming than me.

I like your optimism. If they don't use the integration/sensor, they tend not to bother - low priority from my experience. Your response, @Ernst79, to issues and new sensors is awesome.

**Comment by hugalafutro** [comment]
So far I successfully migrated Govee, Switchbot and MiFlora devices to the new HA BT integration. It's not picking up LYWSD03MMC flashed with ATC firmware or Xiaomi MiScale (yet?), but I'll do those once/if it does.

On the whole doing it few devices at a time as they get detected is not half as bad as I feared.

**Comment by Ernst79** [comment]
I’m currently working on the LYWSD03MMC with custom Firmware, as the first sensor to move. Not sure if I make it before next month release though, as this is the first I do (The others are done by the HA devs, but I have some good examples now. The rest will follow asap, but it will take a couple of releases.

Miscale is a different format, compared to xiaomi MiBeacon. This will follow later as well.

**Reply by fbalesse** [reply]
Hi, my LYWSD03MMC with ATC firmware keep always not compatible with the new integration. Anything new with this ? With original firmware, it is ok, but ATC is better and I prefer keep them.
Thank you.

**Reply by Ernst79** [reply]
It is already compatible, when you use the BTHome format. It’s available in the pvvx firmware as one of the BLE broadcasting formats

**Reply by fbalesse** [reply]
Thank you. It's work.

**Comment by hugalafutro** [comment]
Have to say my enthusiasm for the HA's integration of bluetooth grows colder by the day. CGPR1 Motion sensor has no configuration options and remains triggered for the default 10minutes rendering it unusable, iBeacon trackers never change from Home rendering them unusable.

Everything I try to set up with the new HA bt integration is a massive undertaking, doesn't work straight out of the box, misses basic configuration options only people who actually never use what they program support for could omit and so on and so forth.

Bottom line is after spendning hours messing with converting the devices I end up just adding them incomparably easier into your integration and I'm done in 5 minutes.

Probably gonna have to stop updating HA once/if it drops support for your integration if things don;t change.
